### ISSUE TITLE
Hotspots on Nexus

### ISSUE THREAD

**Issue by xUNTOUCHABLEx** [psforever/PSF-LoginServer#1039]
Hotspots on Nexus are staying and not disappearing after a given amount of time.

**Comment by ScrawnyRonnie** [comment]
There is another side to this issue that I'm not exactly sure how to reproduce. There are invisible hotspots (I assume this is what causes this) that will cause IA to send you to a cont that has none there. On the main server (51000) right now if you IA it takes you to Esamir. I made a new hotspot there and it soon disappeared as it should, but despite no visible ones remaining it thinks there is one here. Maybe our testing wasn't flawed, but there could be a specific action(s) that can cause them to last forever?

If invisible hotspots aren't the right way to describe it, is there anything else that dictates where IA would send someone or does it strictly use hotspots to either work or say there are none?

**Comment by Fate-JH** [comment]
It will tell you if there are no hotspots available.

There are actually two hotspot systems at play.  The first is a shorter-lived one that represents the hotspot that should be visible.  Alongside that is a hotspot history that maintains the memory of the hotspot for a while longer.  The purpose of the history is to ensure that any hotspot is alive long enough so instant action doesn't only work for <60s intervals at a time.

The systems might be leaking over one another.

**Comment by ScrawnyRonnie** [comment]
This one is partially resolved already and was opened before the server was updated with the fix. We're not seeing hotspots that don't go away anymore.

There is still the lingering issue of IA working long after no hotspots exist. It doesn't happen every time though and I'm not sure anyone knows how to make it happen. As far as I know the last time it happened was on the 9th as per my last comment.

**Comment by Fate-JH** [comment]
About how long after a hotspot has decayed and is no longer visible has Instant Action been able to detect it?

**Comment by ScrawnyRonnie** [comment]
The next time it happens I'll stick around for a while to see how long I can continue to IA without creating more.

**Comment by ScrawnyRonnie** [comment]
Okay I figured it out just by random chance. First, it has nothing to do with a hotspot having been somewhere. Here are two ways I have found IA to send you to a continent with no hotspots:

**This only applies to and happens with home conts**; with neutral conts it does not happen.
- If your character is on any home cont and your empire owns a base or tower, IA will send you there. This happens even if the below conditions are also met.
- If you are on a neutral cont or in sanc, an enemy player is on a home cont, and your empire owns a base or tower on that cont, IA will send you to the cont the enemy player is on.
